File: examples/node/openai_example.ts
/**
 * Example: build the OpenAI chat-completions payload for the code-reviewer skill.
 *
 * Run with:
 *   cd packages/usf-ts && npm run build && cd ../..
 *   npx tsx examples/node/openai_example.ts
 *
 * This script stays offline — it prints the payload. To actually call OpenAI,
 * pass `payload` to the official `openai` SDK (or fetch).
 */
import path from "node:path";
import { fileURLToPath } from "node:url";
import { load, getAdapter } from "../../packages/usf-ts/src/index.js";

const here = path.dirname(fileURLToPath(import.meta.url));
const skillPath = path.resolve(here, "../../skills/code-reviewer.md");

async function main() {
  const skill = await load(skillPath);
  const compiled = await skill.buildPrompt({
    code: "function divide(a, b) { return a / b; }",
    language: "javascript",
  });
  const payload = getAdapter("openai").render(compiled);
  console.log(JSON.stringify(payload, null, 2));
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: examples/python/openai_example.py
"""Run the code-reviewer skill against OpenAI.

Prereq:
    pip install -e packages/usf-py[openai]
    export OPENAI_API_KEY=sk-...

Run:
    python examples/python/openai_example.py
"""
from __future__ import annotations

from pathlib import Path

from usf import load

SKILL = Path(__file__).resolve().parents[2] / "skills" / "code-reviewer.md"


def main() -> None:
    skill = load(SKILL)
    output = skill.run(
        "openai",
        {
            "code": "def divide(a, b):\n    return a / b",
            "language": "python",
        },
    )
    print(output)


if __name__ == "__main__":
    main()
